packages/api/src/mocks.ts

```diff
@@ -59,19 +62,88 @@ function mockInternals() {
  * })
  * ```
  *
+ * `listen` can also be mocked with direct calls to the `emit` function:
+ * ```js
+ * import { mockIPC, clearMocks } from "@tauri-apps/api/mocks"
+ * import { emit, listen } from "@tauri-apps/api/event"
+ *
+ * afterEach(() => {
+ *    clearMocks()
+ * })
+ *
+ * test("mocked event", () => {
+ *  mockIPC(() => {}); // a call to `mockIPC` is still required even if no `invoke` calls are mocked
+ *
+ *  const eventHandler = vi.fn();
+ *  listen('test-event', eventHandler); // typically in component setup or similar
+ *
+ *  emit('test-event', { foo: 'bar' });
+ *  expect(eventHandler).toHaveBeenCalledWith({
+ *    event: 'test-event',
+ *    payload: { foo: 'bar' }
+ *  });
+ * })
+ * ```
+ * `emitTo` is currently **not** supported by this mock implementation.
+ *
  * @since 1.0.0
  */
 export function mockIPC(
   cb: (cmd: string, payload?: InvokeArgs) => unknown
 ): void {
   mockInternals()
 
+  function isEventPluginInvoke(cmd: string): boolean {
+    return cmd.startsWith('plugin:event|')
+  }
+
+  function handleEventPlugin(cmd: string, args?: InvokeArgs): unknown {
+    switch (cmd.split('|')[1]) {
+      case 'listen':
+        return handleListen(args as { event: EventName; handler: number })
+      case 'emit':
+        return handleEmit(args as { event: EventName; payload?: unknown })
+      case 'unlisten':
+        return handleRemoveListener(args as { event: EventName; id: number })
+    }
```

Comment:
```suggestion
    switch (cmd) {
      case 'plugin:event|listen':
        return handleListen(args as { event: EventName; handler: number })
      case 'plugin:event|emit':
        return handleEmit(args as { event: EventName; payload?: unknown })
      case 'plugin:event|unlisten':
        return handleRemoveListener(args as { event: EventName; id: number })
    }
```

I know the intention was to reduce duplications, but I think this makes it easier to understand